hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageDispatcherServiceIntegrationTest.java

```diff
@@ -159,6 +162,38 @@ public void sendDeleteMessage() {
         assertDeleteMessage(controllerId);
     }
 
+
+    @Test
+    @Description("Verify that attribute update is requested after device successfully closed software update.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 1),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 2),
+            @Expect(type = ActionUpdatedEvent.class, count = 2), @Expect(type = ActionCreatedEvent.class, count = 2),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 3),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 1),
+            @Expect(type = TargetUpdatedEvent.class, count = 4),
+            @Expect(type = TargetAttributesRequestedEvent.class, count = 1),
+            @Expect(type = TargetPollEvent.class, count = 1) })
+    public void attributeRequestAfterSuccessfulUpdate() {
+        final String controllerId = TARGET_PREFIX + "attributeUpdateRequest";
+        registerAndAssertTargetWithExistingTenant(controllerId);
+        final Target target = controllerManagement.getByControllerId(controllerId).get();
+        final DistributionSet distributionSet = testdataFactory.createDistributionSet(UUID.randomUUID().toString());
+
+        final DistributionSetAssignmentResult assignmentResult1 = assignDistributionSet(distributionSet, target);
```

Comment:
you could get the actionId here right away so you don't need to save the DistributionSetAssignmentResult object